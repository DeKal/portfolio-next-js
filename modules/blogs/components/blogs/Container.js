import styled from 'styled-components/macro'
import Container from 'react-bootstrap/Container'

export default styled(Container)`
  margin-top: 30px;

  @media (min-width: 768px),
    @media (min-width: 992px),
    @media (min-width: 1200px) {
    max-width: ${props => props.theme.postMaxWidth} !important;
  }
`
