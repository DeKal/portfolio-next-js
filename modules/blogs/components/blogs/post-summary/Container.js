import styled from 'styled-components/macro'

export default styled.div`
  margin-bottom: 24px;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.1);
  padding: 23px 0;
  background-color: rgba(255, 255, 255, 1);
`
