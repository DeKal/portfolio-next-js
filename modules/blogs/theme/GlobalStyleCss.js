import React from 'react'
import PropTypes from 'prop-types'

const GlobalStyleCss = () => {
  return (
    <style jsx global>{`
      p {
        margin-top: 1.5em;
        margin-bottom: 5px;
      }
      h1 {
        margin-top: 2em;
      }
      h2,
      h3,
      h4,
      h5,
      h6 {
        margin-top: 1.7em;
      }
    `}</style>
  )
}

GlobalStyleCss.propTypes = {
  theme: PropTypes.any
}

export default GlobalStyleCss
