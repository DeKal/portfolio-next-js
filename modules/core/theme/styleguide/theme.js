import baseTheme from './baseTheme'

const theme = {
  ...baseTheme,
  navHeaderHeight: '40px',
  navContentMinHeight: '80px',
  footerHeight: '50px',
  navHeightWhenScrollingOnMobile: '45px',
  navPosWhenHiding: '-220px',
  aboutMarginTop: '-200px',
  colors: {
    // Color variables
    ghostWhite: '#f8f8ff ',
    colorLight: '#ecf0f1',
    colorDarken: '#bdc3c7',
    colorDark: '#2c3e50',
    colorAccent: '#FFA726',
    colorBlack: '#000000',
    bgColor: '#fff',
    grey: '#696969'
  },
  images: {
    background: '/images/bg.webp',
    backgroundMobile: '/images/bg-mobile.webp',
    bgPattern: '/images/bg-pattern.jpg',
    bgPatternOpacity: '1',
    overlayBgColor: 'rgba(44, 62, 80, 0.4)',
    overlayBgColorMobile: 'rgba(44, 62, 80, 0.3)',
    overlay:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAIGNIUk0AAHolAACAgwAA+f8AAIDoAABSCAABFVgAADqXAAAXb9daH5AAAAAZSURBVHjaYmCAAGM0GjfHGEMGAAAA//8DACm0ATMe8beGAAAAAElFTkSuQmCC'
  },
  productImageRatio: '143%',
  headerPadding: '200px 10px 250px'
}

const themeExtra = {
  ...theme,
  extra: {
    boxShadow: `0 1px 2px -1px ${theme.colors.colorDark}`
  }
}

export default themeExtra
