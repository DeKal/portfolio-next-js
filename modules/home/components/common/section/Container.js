import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

const Container = ({ id, className, children }) => (
  <Style id={id} className={className}>
    {children}
  </Style>
)

Container.propTypes = {
  className: PropTypes.string,
  children: PropTypes.any,
  id: PropTypes.any
}

export default Container

const Style = styled.section`
  position: relative;
  margin-top: 100px;
  padding-top: 25px;
  padding-bottom: 25px;

  @media screen and (max-width: 800px) {
    margin-top: 30px;
  }
`
