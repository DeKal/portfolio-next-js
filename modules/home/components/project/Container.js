import styled from 'styled-components/macro'
import Container from '~/modules/home/components/common/section/Container'

const StyleContainer = styled(Container)`
  background: #fffff2;
`
export default StyleContainer
