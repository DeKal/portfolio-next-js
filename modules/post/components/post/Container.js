import styled from 'styled-components/macro'
import Container from 'react-bootstrap/Container'

export default styled(Container)`
  margin-top: ${props => props.theme.postMarginTop};
  padding: ${props => props.theme.postSurroundingsPadding};
  background-color: ${props => props.theme.colors.light};

  @media (min-width: 768px),
    @media (min-width: 992px),
    @media (min-width: 1200px) {
    max-width: ${props => props.theme.postMaxWidth} !important;
  }
`
