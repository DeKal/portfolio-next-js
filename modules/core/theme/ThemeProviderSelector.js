import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components/macro'
import {
  isChristmasDay,
  isNewYearComing,
  isTet,
  isBlogs
} from '~/modules/core/utils/helpers'
import theme from '~/modules/core/theme/styleguide/theme'
import christMasTheme from '~/modules/core/theme/styleguide/christMasTheme'
import newYearTheme from '~/modules/core/theme/styleguide/newYearTheme'
import tetTheme from '~/modules/core/theme/styleguide/tetTheme'
import blogTheme from '~/modules/core/theme/styleguide/blogTheme'
import BlogGlobalStyleCss from '~/modules/post/theme/GlobalStyleCss'
import HomeGlobalStyleCss from '~/modules/home/theme/GlobalStyleCss'
import { useRouter } from 'next/router'

const ThemeProviderSelector = ({ children }) => {
  const router = useRouter()
  const theme = useTheme(router)
  return (
    <ThemeProvider theme={theme}>
      {children}
      {selectGlobalCss(router, theme)}
    </ThemeProvider>
  )
}

ThemeProviderSelector.propTypes = {
  children: PropTypes.any
}

export default ThemeProviderSelector

const useTheme = router => {
  if (isBlogs(router.pathname)) {
    return blogTheme
  }

  if (isChristmasDay()) {
    return christMasTheme
  }

  if (isNewYearComing()) {
    return newYearTheme
  }

  if (isTet()) {
    return tetTheme
  }

  return theme
}

const selectGlobalCss = (router, theme) => {
  if (!isBlogs(router.pathname)) {
    return <HomeGlobalStyleCss theme={theme} />
  }
  return <BlogGlobalStyleCss theme={theme} />
}
