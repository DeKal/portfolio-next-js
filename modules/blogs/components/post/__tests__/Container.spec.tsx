import * as React from 'react'
import { shallowWithBaseTheme } from '~/modules/core/utils/withThemeProviders'
import Container from '~/modules/blogs/components/post/Container'
import theme from '~/modules/core/theme/styleguide/blogTheme'

describe('Blog Container', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithBaseTheme(<Container />, theme)
    expect(wrap).toMatchSnapshot()
  })
})
