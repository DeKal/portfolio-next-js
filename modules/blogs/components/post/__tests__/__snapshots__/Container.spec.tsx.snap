// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Container should render without throwing an error 1`] = `
<Container
  className="Container-ffzdp6-0 gIFPgt"
  fluid={false}
/>
`;
